react/ConfirmationMessage.tsx

```diff
@@ -1,27 +1,50 @@
-import React, { FC } from 'react'
+import React, { FC, useEffect, useState } from 'react'
 import { FormattedMessage } from 'react-intl'
 import { useCssHandles } from 'vtex.css-handles'
 
 import { useOrderGroup } from './components/OrderGroupContext'
+import { useSessionResponse } from './utils/useSessionResponse'
 
 const CSS_HANDLES = ['confirmationMessage']
 
-const ConfirmationMessage: FC = () => {
+interface Props {
+  enableGuestMessage: boolean
+}
+
+const ConfirmationMessage: FC<Props> = ({
+  enableGuestMessage = false,
+}: Props) => {
   const handles = useCssHandles(CSS_HANDLES)
   const orderGroup = useOrderGroup()
   const profile = orderGroup.orders[0].clientProfileData
+  const sessionResponse = useSessionResponse()
+  const [isLoggedIn, setIsLoggedIn] = useState(true)
+
+  useEffect(() => {
+    setIsLoggedIn(!!sessionResponse?.namespaces?.profile?.email)
+  }, [sessionResponse, setIsLoggedIn])
 
   return (
     <p
       className={`${handles.confirmationMessage} mt5 t-body tc c-muted-1 lh-copy`}
     >
-      <FormattedMessage
-        id="store/header.email"
-        values={{
-          lineBreak: <br />,
-          userEmail: <strong className="nowrap">{profile.email}</strong>,
-        }}
-      />
+      {!isLoggedIn && enableGuestMessage ? (
+        <FormattedMessage
+          id="store/header.guest-email"
+          values={{
+            lineBreak: <br />,
+            userEmail: <strong className="nowrap">{profile.email}</strong>,
+          }}
+        />
+      ) : (
+        <FormattedMessage
+          id="store/header.email"
+          values={{
+            lineBreak: <br />,
+            userEmail: <strong className="nowrap">{profile.email}</strong>,
+          }}
+        />
+      )}
```

Comment:
I noticed that both `store/header.guest-email` and `store/header.email` keys are using the same string value. Since the component is set up to conditionally render one of these based on `isLoggedIn` and `enableGuestMessage`, I was wondering if there’s a specific scenario where these two messages are intended to be different?

As it currently stands, it seems that the same text will be displayed regardless of the user's login state. Could you clarify the objective here? Is the intention to eventually have different messages for logged-in and guest users?

Reply:
Yes, we duplicated the message for now just to enable having two different Ids so the merchant is able to differentiate those messages using the Graphql Layer when needed. 
If the PT has already an idea to differentiate this message we could open a separate PR since this will need to be sent to the localization team. What do you think?

Reply:
@beatrizmaselli thanks for your answer!

Reply:
@wagnerlduarte yes, as Beatiz said by default these two messages are same, but customer can assign their own text via GraphQL. I've added a comment in the code to clarify it! Thank you!